components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/security/handler/BasicSecurityHandler.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis.security.handler;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.CarbonException;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.securevault.SecretResolver;
+import org.wso2.securevault.SecretResolverFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.Objects;
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * This class extends the SecurityHandlerAdapter to create a basic security handler with a user store defined in
+ * internal-apis.xml.
+ */
+public class BasicSecurityHandler extends SecurityHandlerAdapter {
+
+    private static final Log LOG = LogFactory.getLog(BasicSecurityHandler.class);
+
+    private String name;
+    private SecretResolver secretResolver;
+    private OMElement usersOM = null;
+
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    @Override
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    protected Boolean authenticate(String credentials) {
+
+        String decodedCredentials = new String(new Base64().decode(credentials.getBytes()));
+        String userNameFromHeader = decodedCredentials.split(":")[0];
+        String passwordFromHeader = decodedCredentials.split(":")[1];
+        if (usersOM == null) {
+            setUserStore();
+        }
+        if (Objects.nonNull(usersOM)) {
+            Iterator userList = usersOM.getChildrenWithName(new QName("user"));
+            if (Objects.nonNull(userList)) {
+                String passwordFromStore = null;
+                String userNameFromStore;
+                if (isValid(passwordFromHeader) && isValid(userNameFromHeader)) {
+                    while (userList.hasNext()) {
```

Comment:
Let's load and store users in a data structure instead of going through OMElement